src/rgw/rgw_sal_motr.cc

```diff
@@ -2219,7 +2220,18 @@ int MotrObject::remove_mobj_and_index_entry(
         }
       }
       size_rounded = roundup(ent.meta.size, get_unit_sz());
-      rc = this->delete_mobj(dpp);
+      if (store->gc_enabled()) {
+        std::string tag = PRIx64 + std::to_string(this->meta.oid.u_hi) + ":" +
+                          PRIx64 + std::to_string(this->meta.oid.u_lo);
+        std::string obj_fqdn = bucket_name + delete_key;
+        ::Meta *mobj = reinterpret_cast<::Meta*>(&this->meta);
+        motr_gc_obj_info obj(tag, obj_fqdn, *mobj, std::time(nullptr),
+                             ent.meta.size, size_rounded, false, "");
+        rc = store->get_gc()->enqueue(obj);
+        pushed_to_gc = (rc == 0);   
+      }
+      if (! pushed_to_gc)
```

Comment:
if (! pushed_to_gc) evaluates to false, you can log a message indicating object (oid, pv id, layout id) would be deleted by Motr GC. Would be useful by QA for validation.

Reply:
addressed. 👍